CONTRIBUTING.md

```diff
@@ -0,0 +1,412 @@
+CONTRIBUTING
+============
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+
+- [Getting Started](#getting-started)
+  - [Clone & Install](#clone-&-install)
+  - [Create the Component](#create-the-component)
+  - [Open A PR](#open-a-pr)
+  - [Commands](#commands)
+- [Writing A Component](#writing-a-component)
+  - [_meta](#_meta)
+  - [API](#api)
+    - [SUI HTML Classes](#sui-html-classes)
+    - [SUI HTML Markup](#sui-html-markup)
+  - [Testing](#testing)
+    - [Common Tests](#common-tests)
+  - [State](#state)
+    - [AutoControlledComponent](#autocontrolledcomponent)
+- [Documentation](#documentation)
+  - [Website](#website)
+  - [Components](#components)
+  - [Props](#props)
+  - [Examples](#examples)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Getting Started
+
+### Clone & Install
+
+Start by cloning this repo and installing dependencies:
+
+```sh
+git clone git@github.com:TechnologyAdvice/stardust.git
+cd stardust
+npm install
+```
+
+### Create the Component
+
+Create components in `src`.  The directory structure follows SUI naming conventions.
+
+Stateless components should be written as a `function`:
+
+```js
+function Button(props) {
+  // ...
+}
+```
+
+Stateful components should be classes:
+ 
+```js
+import AutoControlledComponent from 'src/utils/AutoControlledComponent'
+
+class Dropdown extends AutoControlledComponent {
+  // ...
+}
+```
+
+You probably need to extend our [`AutoControlledComponent`](#autocontrolledcomponent) to support both [controlled][2] and [uncontrolled][3] component patterns.
+
+### Open A PR
+
+This is a good time to open your PR.  The component has been created, but the API and internals are not yet coded.
+
+### Commands
+
+This list is not updated, you should run `npm run` to see all scripts.
+
+```sh
+npm start                     // run doc site
+npm run start:local-modules   // run offline (slower builds)
+
+npm test                      // test once
+npm run test:watch            // test on file change
+
+npm run build                 // build everything
+npm run build:dist            // build dist 
+npm run build:docs            // build docs
+npm run build:docs-toc        // build toc for markdown files
+
+npm run deploy:docs           // deploy gh-pages doc site 
+
+npm run lint                  // lint once
+npm run lint:fix              // lint and attempt to fix
+npm run lint:watch            // lint on file change
+```
+
+## Writing A Component
+
+### _meta
+
+Every component has a static property called `_meta`.  This object defines the component.  The values here are used in `propTypes`, generated documentation, generated test cases, and some utilities.
```

Comment:
Oh gross, the markdown is messed up in the diff from the `_meta`. Looks fine in the real renderer, though.